test/pollinate_tests.ml

```diff
@@ -5,67 +5,122 @@ module Client_tests = struct
     | Success of string
-    | Error of string
-  [@@deriving bin_io]
+    | Error   of string
+  [@@deriving bin_io, show { with_path = false }]
 
   type state = string list
 
   let msg_handler state _ request =
     let request = Util.Encoding.unpack bin_read_request request in
     let response =
       match request with
-      | Get -> List (!state)
+      | Ping -> Pong
+      | Get -> List !state
       | Insert s ->
         state := s :: !state;
         Success "Successfully added value to state" in
     Util.Encoding.pack bin_writer_response response
 
-    (* Initializes two peers and has each one request the state
-    of the other, returning the first element in the response of each *)
+  let client_a =
+    Lwt_main.run @@ Client.init ~state:["test1"] ~msg_handler ("127.0.0.1", 3000)
+  let client_b =
+    Lwt_main.run @@ Client.init ~state:["test2"] ~msg_handler ("127.0.0.1", 3005)
+
+  (* Initializes two peers and has each one request the state
+     of the other, returning the first element in the response of each *)
   let trade_messages () =
-    let%lwt client_a = Client.init ~state:["test1"] ~msg_handler ("127.0.0.1", 3000) in
     let peer_a = Client.peer_from !client_a in
 
-    let%lwt client_b = Client.init ~state:["test2"] ~msg_handler ("127.0.0.1", 3005) in
     let peer_b = Client.peer_from !client_b in
 
-    let req = Util.Encoding.pack bin_writer_request Get in
+    let get = Util.Encoding.pack bin_writer_request Get in
 
-    let%lwt () = Client.send_to client_a req peer_b in
-    let%lwt (res_a, _) = Client.recv_next client_a in
+    let%lwt () = Client.send_to client_a get peer_b in
+    let%lwt res_a, _ = Client.recv_next client_a in
 
     let res_a = Util.Encoding.unpack bin_read_response res_a in
 
-    let%lwt () = Client.send_to client_b req peer_a in
-    let%lwt (res_b, _) = Client.recv_next client_b in
+    let%lwt () = Client.send_to client_b get peer_a in
+    let%lwt res_b, _ = Client.recv_next client_b in
 
     let res_b = Util.Encoding.unpack bin_read_response res_b in
 
-    let (res_a, res_b) =
+    let res_a, res_b =
       match (res_a, res_b) with
-      | (List l1, List l2) -> (List.hd l1, List.hd l2)
+      | List l1, List l2 -> (List.hd l1, List.hd l2)
       | _ -> failwith "Incorrect response" in
 
     Lwt.return (res_a, res_b)
+  let test_insert () =
+    let _ = Client.peer_from !client_a in
+
+    let peer_b = Client.peer_from !client_b in
+
+    let req = Util.Encoding.pack bin_writer_request @@ Insert "something" in
+
+    let%lwt () = Client.send_to client_a req peer_b in
+    let%lwt res_a, _ = Client.recv_next client_a in
+
+    let res_a = Util.Encoding.unpack bin_read_response res_a in
+
+    let get = Util.Encoding.pack bin_writer_request Get in
+
+    let%lwt () = Client.send_to client_a get peer_b in
+    let%lwt status_of_b, _ = Client.recv_next client_a in
+
+    let status_of_b = Util.Encoding.unpack bin_read_response status_of_b in
+
+    let res_a, status_of_b =
```

Comment:
This test is failing, I don't know what I miss, but I expect `status_of_b` to be `something` (the inserted value provided line 67).
Would you mind help me to figure our what I am missing?